Source/Src_nd/make_edge_scal.F90

```diff
@@ -133,145 +154,153 @@ subroutine make_edge_scal_2d(domlo, domhi, lo, hi, &
 
     ! loop over appropriate x-faces
     if (ppm_type .eq. 0) then
-       do j=js-1,je+1
-          do i=is,ie+1
+       do j=lo(2)-1,hi(2)+1
+          do i=lo(1),hi(1)+1
              ! make slx, srx with 1D extrapolation
-             slx(i,j) = s(i-1,j,comp) + (HALF - dt2*umac(i,j)/hx)*slopex(i-1,j,1)
-             srx(i,j) = s(i  ,j,comp) - (HALF + dt2*umac(i,j)/hx)*slopex(i  ,j,1)
+             slx(i,j,k) = s(i-1,j,k,comp) + (HALF - dt2*umac(i,j,k)/hx)*slopex(i-1,j,k,1)
+             srx(i,j,k) = s(i  ,j,k,comp) - (HALF + dt2*umac(i,j,k)/hx)*slopex(i  ,j,k,1)
           enddo
        enddo
     else if (ppm_type .eq. 1 .or. ppm_type .eq. 2) then
-       do j=js-1,je+1
-          do i=is,ie+1
+       do j=lo(2)-1,hi(2)+1
+          do i=lo(1),hi(1)+1
              ! make slx, srx with 1D extrapolation
-             slx(i,j) = Ip(i-1,j,1)
-             srx(i,j) = Im(i  ,j,1)
+             slx(i,j,k) = Ip(i-1,j,k,1)
+             srx(i,j,k) = Im(i  ,j,k,1)
           end do
        end do
     end if
 
     ! impose lo side bc's
     if (lo(1) .eq. domlo(1)) then
        if (adv_bc(1,1,bccomp) .eq. EXT_DIR) then
-          slx(is,js-1:je+1) = s(is-1,js-1:je+1,comp)
-          srx(is,js-1:je+1) = s(is-1,js-1:je+1,comp)
+          slx(lo(1),lo(2)-1:hi(2)+1,k) = s(lo(1)-1,lo(2)-1:hi(2)+1,k,comp)
+          srx(lo(1),lo(2)-1:hi(2)+1,k) = s(lo(1)-1,lo(2)-1:hi(2)+1,k,comp)
        else if (adv_bc(1,1,bccomp) .eq. FOEXTRAP .or. &
             adv_bc(1,1,bccomp) .eq. HOEXTRAP) then
           if (is_vel .eq. 1 .and. comp .eq. 1) then
-             srx(is,js-1:je+1) = min(srx(is,js-1:je+1),0.d0)
+             srx(lo(1),lo(2)-1:hi(2)+1,k) = min(srx(lo(1),lo(2)-1:hi(2)+1,k),0.d0)
           end if
-          slx(is,js-1:je+1) = srx(is,js-1:je+1)
+          slx(lo(1),lo(2)-1:hi(2)+1,k) = srx(lo(1),lo(2)-1:hi(2)+1,k)
        else if (adv_bc(1,1,bccomp) .eq. REFLECT_EVEN) then
-          slx(is,js-1:je+1) = srx(is,js-1:je+1)
+          slx(lo(1),lo(2)-1:hi(2)+1,k) = srx(lo(1),lo(2)-1:hi(2)+1,k)
        else if (adv_bc(1,1,bccomp) .eq. REFLECT_ODD) then
-          slx(is,js-1:je+1) = 0.d0
-          srx(is,js-1:je+1) = 0.d0
+          slx(hi(1)+1,lo(2)-1:hi(2)+1,k) = 0.d0
```

Comment:
I think the `i` index is supposed to be `lo(1)` here.

Reply:
The array `slx` is replaced by a scalar variable in #81, so ultimately once everything has been merged this won't matter, but yes that should be a `lo(1)`